### PR TITLE
Proxy server implemented

### PR DESCRIPTION
**Pull request by hyunsikjeong**
Implemented a proxy server.
Check the arguments with `-h` flag, or read the source code.

### CODE REVIEW

**Review by sgkim126 — COMMENTED** [review]
Please add requirements.txt

**Review by sgkim126 — CHANGES_REQUESTED** [review]
Please add requirements.txt

**Comment by hyunsikjeong** [comment]
Added [requirements.txt](https://github.com/jhs7jhs/codechain-proxy/blob/ba6df429df11ab98064131490c35280ff3531de2/requirements.txt).

File: proxy.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
```
Comment (sgkim126) [inline proxy.py]: How about python3?

File: proxy.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    parser.add_argument('--port', type=int,
+                        help='binding port')
+    parser.add_argument('--forward', type=int, default=8080,
+                        help='port to forward (deafult: 8080)')
+
+    args = parser.parse_args()
+
+    app = Flask(__name__)
+    with open(args.whitelist) as f:
+        app.whitelist = set(line for line in f.readlines())
```
Comment (sgkim126) [inline proxy.py]: It must be
```python
with open(args.whitelist) as f:
    app.whitelist = set(line.rstrip() for line in f)
```

File: proxy.py (hunk 1 above)
Comment (hyunsikjeong) [reply proxy.py]: I changed it to the `python3`. [Link](https://github.com/jhs7jhs/codechain-proxy/blob/ef86a16b71e37ce0612fb3d4741e701e721fee91/proxy.py)

File: proxy.py (hunk 2 above)
Comment (sgkim126) [reply proxy.py]: The current code doesn't remove the new line.

**Review by sgkim126 — CHANGES_REQUESTED** [review]

File: proxy.py (hunk 2 above)
Comment (hyunsikjeong) [reply proxy.py]: Fixed it.

**Review by sgkim126 — APPROVED** [review]

**Pull request merged**